demo/src/demo_agent.rs

```diff
@@ -0,0 +1,275 @@
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use env_logger;
+use serde::Deserialize;
+use tokio::sync::mpsc::Receiver;
+use tokio::sync::Mutex;
+use url::Url;
+
+use aries_vcx::handlers::out_of_band::sender::OutOfBandSender;
+use aries_vcx::handlers::util::AnyInvitation;
+use aries_vcx::messages::msg_fields::protocols::out_of_band::invitation::OobService;
+use aries_vcx::protocols::connection::invitee::states::completed::Completed;
+use aries_vcx::protocols::connection::invitee::states::initial::Initial;
+use aries_vcx::protocols::connection::invitee::states::requested::Requested as InviteeRequested;
+use aries_vcx::protocols::connection::invitee::InviteeConnection;
+use aries_vcx::protocols::connection::inviter::states::requested::Requested as InviterRequested;
+use aries_vcx::protocols::connection::inviter::InviterConnection;
+use aries_vcx::protocols::connection::pairwise_info::PairwiseInfo;
+use aries_vcx::transport::Transport;
+use aries_vcx::utils::mockdata::profile::mock_ledger::MockLedger;
+use aries_vcx_core::ledger::base_ledger::IndyLedgerRead;
+use aries_vcx_core::wallet::base_wallet::BaseWallet;
+use aries_vcx_core::wallet::indy::wallet::create_and_open_wallet;
+use aries_vcx_core::wallet::indy::{IndySdkWallet, WalletConfig};
+use diddoc_legacy::aries::service::AriesService;
+use messages::msg_fields::protocols::basic_message::{BasicMessage, BasicMessageContent, BasicMessageDecorators};
+use messages::msg_fields::protocols::connection::request::Request;
+use messages::msg_fields::protocols::connection::response::Response;
+use messages::msg_fields::protocols::out_of_band::invitation::Invitation;
+use messages::msg_parts::MsgParts;
+use messages::AriesMessage;
+use simple_message_relay::mpsc_registry::MpscRegistry;
+use simple_message_relay::UserMessage;
+
+use crate::http_client::HttpClient;
+
+pub struct DemoAgent {
+    pub wallet: Arc<dyn BaseWallet>,
+    pub endpoint_url: Url,
+    receiver: Arc<Mutex<Receiver<UserMessage>>>,
+    pub name: String,
+    pub log_color: String,
+}
+
+fn build_basic_message(text_message: String) -> BasicMessage {
+    let bm = BasicMessageContent {
+        content: text_message,
+        sent_time: Default::default(),
+    };
+    let bmd = BasicMessageDecorators {
+        l10n: None,
+        thread: None,
+        timing: None,
+    };
+    MsgParts::with_decorators(uuid::Uuid::new_v4().to_string(), bm, bmd)
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct MessageData {
+    message: String,
+    recipient_verkey: String,
+    sender_verkey: String,
+}
```

Comment:
Isn't there some `EncryptionEnvelope` that handles this same thing? I'm in favor of rewriting that, don't get me wrong 😅 , but I just wanted to point out that this might be useful in more places.

Reply:
Yeah it's similar, I think something like this along with `decrypt_as_msg` could be perhaps adopted in `aries-vcx` (with some changes, surely)